src/server/cmd/pachctl/shell/completions.go

```diff
@@ -36,23 +76,19 @@ func RepoCompletion(_, text string, maxCompletions int64) []prompt.Suggest {
 			Description: fmt.Sprintf("%s (%s)", ri.Description, units.BytesSize(float64(ri.SizeBytes))),
 		})
 	}
-	return result
+	return result, samePart(filePart(text))
 }
 
 // BranchCompletion completes branch parameters of the form <repo>@<branch>
-func BranchCompletion(flag, text string, maxCompletions int64) []prompt.Suggest {
-	c, err := client.NewOnUserMachine("user-completion")
-	if err != nil {
-		log.Fatal(err)
-	}
-	defer c.Close()
+func BranchCompletion(flag, text string, maxCompletions int64) ([]prompt.Suggest, CacheFunc) {
+	c := getPachClient()
 	partialFile := cmdutil.ParsePartialFile(text)
+	part := filePart(text)
```

Comment:
`part` here shadows the `part` type, which (while I'm guilty of the same elsewhere) is a little confusing - might be better to rename the type as `partEnum` or something idk w/e

Reply:
Done.